exercise-1.js

```diff
@@ -444,74 +711,148 @@
 
         function maxNumber(numbers) {
-            //your code...
+            let onlyNumbers = [];
+            //console.log(numbers);
+            for (i=0; i < numbers.length; i++) {
+                if (typeof numbers[i] == 'number') {
+                    onlyNumbers.push(numbers[i]);
+                } else if (numbers[i] === "one") {
+                    onlyNumbers.push(1);
+                } else if (numbers[i] === "two") {
+                    onlyNumbers.push(2);
+                } else if (numbers[i] === "three") {
+                    onlyNumbers.push(3);
+                } else if (numbers[i] === "3"){
+                    onlyNumbers.push(3);                    
+                }
+            }
+            return onlyNumbers;
         }
+        let justNumbers = maxNumber(numbersMixed); // NEED TO FINISH
+        console.log(Math.max(...justNumbers));
 
         //After the numbers array has been cleaned up to only have numbers in it, Write a function that sorts the modified numbers array.  Allow the function to sort the array in descending order as well.
 
         function sortNums(numbers,desc=false) {
-            //your code...
+            numbers.sort(function(a, b){return a-b});
+            numbers.sort(function(a, b){return b-a});
         };
+        sortNums(justNumbers);
 
 
 
         /************************************************************* */
         //Research a new feature of ES6+ and create an example of it's use case here.  Be sure to write comments explaining what the feature is and why it is useful.
-
-
             //your code...
-
-
-
+            //ES7
+            // this is the object.entries method. It takes in objects as an argument (Object.entries(CARS)) and truns it into an array of arrays. 
+            // this is useful because it allows you to be able to use the map function with it
+            // the map function takes the entries and maps it for an object.
+            let cars = {
+                honda: 5,
+                mazda: 6,
+                chevy: 7,
+                ford: 9 //additionally, you can add a comma here for the last element. But you dont have too.
+            }
+            for (let [key, val] of Object.entries(cars)) {
+                console.log(key, val);
+            }
         /************************************************************* */
         //Add an example of the 5 primary JavaScript data types to the given mapObj.  The key is an example data type, and the value is the name of the data type.  An object data type has already been set as the 1st key / val pair.
 
         const mapObj = new Map();
-        mapObj.set({company : "TEKsystems"},"object");
-
-        console.log(mapObj.has({company : "TEKsystems"}));
-
+        mapObj.set('company', "TEKsystems");
+        mapObj.set('number', 5);
+        mapObj.set('string', "this is a string");
+        mapObj.set('boolean', true);
+        mapObj.set("array", [1,2,3,4,"hello"]);
+
```

Comment:
These were supposed to be set with the various objects (a number, a string, a boolean, an array) as the 'keys', and the descriptions of those objects ('number', 'string', 'boolean', 'array') as the values.